### PR TITLE
add (installer for) gatk-protected (full, license-restricted GATK)

### PR DESCRIPTION
**Pull request by tomkinsc**
The license for the full version of GATK prevents it from being redistributed. This adds a recipe for a GATK package that provides the commands “gatk” and “gatk-register”. The former is a thin wrapper around the gatk.jar file, and the latter is an installer that copies a user-supplied GenomeAnalysisTK.jar file (or unpacked tar.bz2 file) to the conda environment. It includes helpful messages that instruct the user on what to do.

### CODE REVIEW

**Comment by tomkinsc** [comment]
@bioconda/core, please look over this recipe and let me know what you think.
This address #1247, based on an initial [description](https://github.com/bioconda/bioconda-recipes/issues/1247#issuecomment-208547450) by @johanneskoester.

**Comment by chapmanb** [comment]
Chris -- nice one. Thanks for doing this. The only thing I noticed was that GATK 3.6 is java 8 friendly so we can remove the pin on java-jdk in the requirements. Thanks again.

**Comment by tomkinsc** [comment]
Nice catch, thanks. I've changed it to allow java-jdk >=7.

**Comment by johanneskoester** [comment]
Great work, this is exactly what we needed!

File: recipes/gatk/build.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+mkdir -p $PACKAGE_HOME
+
+cp $RECIPE_DIR/gatk.sh $PACKAGE_HOME/gatk.sh
+
+SOURCE_FILE=$RECIPE_DIR/gatk-register.sh
+DEST_FILE=$PACKAGE_HOME/gatk-register.sh
+
+
+echo "#!/bin/bash" > $DEST_FILE
+echo "PKG_NAME=$PKG_NAME" >> $DEST_FILE
```
Comment (tomkinsc) [inline recipes/gatk/build.sh]: Is there a better way to get the conda package variables into a non-build script?

Comment (johanneskoester) [inline recipes/gatk/build.sh]: At least for me that looks ok.

**Comment by tomkinsc** [comment]
OK to merge?

**Comment by xguse** [comment]
This is seriously great. Thanks for your time on this!

**Comment by yhoogstrate** [comment]
The gatk3-register message refers to the main download page that only shows GATK4. Would it make sense to refer to: https://software.broadinstitute.org/gatk/download/archive ?

**Comment by tomkinsc** [comment]
@yhoogstrate: Yes, that makes sense. I've opened #14989 to change the URL.

**Pull request merged**